Core.ApplicationServices/UserService.cs

```diff
@@ -280,5 +289,51 @@ public void Dispose()
                     )
                 );
         }
+
+        public Maybe<OperationError> DeleteUserFromKitos(Guid userUuid)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var user = _userRepository.AsQueryable().ByUuid(userUuid);
+            if (user == null)
+                return Maybe<OperationError>.Some(new OperationError(OperationFailure.NotFound));
+            if(_activeUserIdContext.GetValueOrDefault().ActiveUserId == user.Id)
+                return Maybe<OperationError>.Some(new OperationError("You cannot delete a user you are currently logged in as", OperationFailure.Forbidden));
+
+
+            if (!_authorizationContext.AllowDelete(user))
+                return Maybe<OperationError>.Some(new OperationError(OperationFailure.Forbidden));
```

Comment:
no need to add Maybe.Some.. returning a new operation error will do the wrapping itself